### PR TITLE
Add Claude Code GitHub Workflow

### PR DESCRIPTION
**Pull request by squadgazzz**
## 🤖 Installing Claude Code GitHub App

This PR adds a GitHub Actions workflow that enables Claude Code integration in our repository.

### What is Claude Code?

[Claude Code](https://claude.com/claude-code) is an AI coding agent that can help with:
- Bug fixes and improvements  
- Documentation updates
- Implementing new features
- Code reviews and suggestions
- Writing tests
- And more!

### How it works

Once this PR is merged, we'll be able to interact with Claude by mentioning @claude in a pull request or issue comment.
Once the workflow is triggered, Claude will analyze the comment and surrounding context, and execute on the request in a GitHub action.

### Important Notes

- **This workflow won't take effect until this PR is merged**
- **@claude mentions won't work until after the merge is complete**
- The workflow runs automatically whenever Claude is mentioned in PR or issue comments
- Claude gets access to the entire PR or issue context including files, diffs, and previous comments

### Security

- Our Anthropic API key is securely stored as a GitHub Actions secret
- Only users with write access to the repository can trigger the workflow
- All Claude runs are stored in the GitHub Actions run history
- Claude's default tools are limited to reading/writing files and interacting with our repo by creating comments, branches, and commits.
- We can add more allowed tools by adding them to the workflow file like:

```
allowed_tools: Bash(npm install),Bash(npm run build),Bash(npm run lint),Bash(npm run test)
```

There's more information in the [Claude Code action repo](https://github.com/anthropics/claude-code-action).

After merging this PR, let's try mentioning @claude in a comment on any PR to get started!

### CODE REVIEW

File: .github/workflows/claude-code-review.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+            ## Related issues
+            <!-- Link related issues here, e.g., Fixes #123 -->
+            EOF
+            ```
+
+            PART 2 - CODE REVIEW:
+
+            REVIEW FOCUS (issues only, no positive feedback):
+            1. **Bugs & Logic Errors**: Race conditions, off-by-one errors, incorrect error handling, panic risks
```
Comment (jmg-duarte) [inline .github/workflows/claude-code-review.yml:67]: Add possible deadlocks (such as holding a regular mutex over an await), unexpected drops (like the request id issue)

File: .github/workflows/claude-code-review.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            2. **Non-Idiomatic Rust**: Unnecessary clones, missing iterators, improper lifetimes, unwrap() abuse
+            3. **Style Deviations**: Check CLAUDE.md for project conventions (alloy vs ethcontract patterns, RPC batching, quote system)
```
Comment (jmg-duarte) [inline .github/workflows/claude-code-review.yml:69]: Btw alloy has a "base" prompt, should we add it or something?

File: .github/workflows/claude-code-review.yml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            4. **Test Coverage**: Missing tests for new functionality, edge cases not covered
```
Comment (jmg-duarte) [inline .github/workflows/claude-code-review.yml:70]: Maybe tell him to suggest tests too instead of just flagging

File: .github/workflows/claude-code-review.yml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+          GH_TOKEN: ${{ github.token }}
+        with:
+          claude_code_oauth_token: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}
+          prompt: |
+            REPOSITORY: ${{ github.repository }}
+            PR NUMBER: ${{ github.event.pull_request.number }}
+            PR AUTHOR: ${{ github.event.pull_request.user.login }}
+            PR TITLE: ${{ github.event.pull_request.title }}
+
+            You are a Rust code reviewer for the CoW Protocol services codebase.
```
Comment (jmg-duarte) [inline .github/workflows/claude-code-review.yml:38]: Does the "experience Rust code reviewer with background in ..." still help?

File: .github/workflows/claude-code-review.yml [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+            TASK:
+            1. Generate a PR description if the PR body is empty
+            2. Review the PR diff and leave inline comments on problematic code
```
Comment (jmg-duarte) [inline .github/workflows/claude-code-review.yml:42]: I've had success telling him to review and sumarize first, do a todo list and then start working, could that be helpful here?

File: .github/workflows/claude-code-review.yml (hunk 4 above)
Comment (m-sz) [reply .github/workflows/claude-code-review.yml:38]: It should. Reassuring AI agents leads to measurably better results.

File: .github/workflows/claude-code-review.yml (hunk 1 above)
Comment (m-sz) [reply .github/workflows/claude-code-review.yml:67]: Yeah, I think deadlocks as a category would be helpful.

File: .github/workflows/claude-code-review.yml [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+permissions:
+  contents: read
+  pull-requests: write
+  issues: read
+  id-token: write
+  actions: read
+
+jobs:
+  claude-review:
+    if: github.event.pull_request.user.login == 'squadgazzz'
```
Comment (m-sz) [inline .github/workflows/claude-code-review.yml:16]: This should be removed to allow everyone's PR to be reviewed

File: .github/workflows/claude-code-review.yml [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+            ## Changes
+            - [List of specific changes made]
+            - [Each bullet should be a concrete change]
+
+            ## How to test
+            [How these changes should be tested. If no new tests were added, write "Existing tests"]
+
+            ## Related issues
+            <!-- Link related issues here, e.g., Fixes #123 -->
```
Comment (m-sz) [inline .github/workflows/claude-code-review.yml:60]: Should we allow `gh pr list` and `gh issue list`, for Claude to access other PR's so they can be referenced here more easily?

**Comment by squadgazzz** [comment]
@m-sz @jmg-duarte let's avoid wasting anyone's time by reviewing draft PRs. This one is not even close to being ready for review.
